README.md

```diff
@@ -107,6 +107,19 @@ The first thing that we need to do is to declare the unique [resource](https://d
       - Keep **State** as **Enabled**
       - Click on the **Add scope** button on the bottom to save this scope.
       - Record the scope's URI (api://{clientid}/access_as_user) for later reference.
+1. Optional: If your application expects to have users with varying access levels, then you can further secure the API by assigning groups and users
+    - If you do not have some already, create a group in Azure
+        - Navigate to the `Azure Active Directory` service, and select the `Groups` page
+        - Select `New Group` at the top, and create an appropriate group for this sample
+            - Name it something meaningful, such as `java-webapi-group`, as well as a description for what the group is used for
+            - In the `Members` section, add any accounts you plan to use when testing this sample
+    - Once you have a group, you can assign it to your application and restrict access to only members of that group
+        - Navigate to the `Azure Active Directory` service, and select the `Enterprise Applications` page
+        - Find and select the application you created (`Java-webapi`) in the list shown
```

Comment:
This would allow users to limit access to a whole application (in this case 'Java-webapi') by using groups, but would not let customers limit access per endpoint, right? 

I think we should support the scenario where have webapi.com/users and webapi.com/admin, and webapi.com/users can be accessed by group "users", but webapi.com/admin can only be accessed by group "admin". By supporting this scenario, we by default support the scenario where we want limit access to the whole application.

Reply:
@sangonzal : This was discussed at our standups, but the gist of the discussion was that there were two main parts to demo-ing group claims: how to set them in Azure and see them in a token fetched by the library, and how to use the groups from the tokens to restrict specific web pages (like you describe). 

The first part is clearly MSAL/Azure related so it made sense to add that into the sample, but the second part would basically just be a Spring tutorial about creating custom roles and token filters, managing user detail objects, etc., none of which is unique to MSAL. We figured it'd be better to keep things focused on MSAL rather then bloating the amount of Spring-specific in the tutorial, and assume users already have some way of managing users and their roles.

However, I agree that this isn't really a full demo of group claims, and users could have trouble actually using the groups in their app/outside of Azure. As @idg-sam said, having an incremental tutorial could be a good idea: keep this one as-is as a demo for group claims basics (and probably add a page to the wiki on the main repo), and create a new full "how to use group claims in Spring/in a Java servlet" sample, with a link to this sample and/or the wiki?

Reply:
Adding a section to the wiki is a great idea. May I suggest adding links to 
- https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-app-manifest#groupmembershipclaims-attribute
- https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-app-manifest#groupmembershipclaims-attribute